prediction_market_agent/agents/microchain_agent/agent_functions.py

```diff
@@ -0,0 +1,55 @@
+import typing as t
+
+from microchain import Agent, Function
+
+from prediction_market_agent.agents.microchain_agent.prompts import (
+    NON_UPDATABLE_DIVIDOR,
+    SYSTEM_PROMPT,
+)
+
+
+class AgentAction(Function):
+    def __init__(self, agent: Agent) -> None:
+        self.agent = agent
```

Comment:
These functions allow to change the agent itself

Reply:
Something about this doesn't feel right. Intuitively I would say that an agent should have a Function. But not the other way around. Can't think of another way of doing this with microchain though...

Reply:
> Intuitively I would say that an agent should have a Function. But not the other way around

That's how crewai/autogen have it:

```
        llm_config["tools"] = [google_search_tool.schema, web_scaping_tool.schema]
        self.assistant = GPTAssistantAgent(
            name="assistant",
            instructions="...",
            llm_config=llm_config,
        )
```
so I wonder if this kind of behaviour is possible with these frameworks.

Reply:
Yep, it feels odd; a function like this can't even come from the marketplace. 

So if you have a better idea how to implement this I'm open to it 😄 Maybe the agent class could have something like `get_my_functions(self) ->? list[AgentFunction]` method?

Reply:
> Maybe the agent class could have something like get_my_functions(self) -> list[AgentFunction] method

Yes that sounds like a good idea. Probably not high priority, so I'll just make a task to think about it more for now https://github.com/gnosis/prediction-market-agent/issues/253